crypto/default/src/main/java/org/keycloak/crypto/hash/Argon2PasswordHashProviderFactory.java

```diff
@@ -0,0 +1,47 @@
+package org.keycloak.crypto.hash;
+
+import org.keycloak.Config;
+import org.keycloak.credential.hash.PasswordHashProvider;
+import org.keycloak.credential.hash.PasswordHashProviderFactory;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+
+public class Argon2PasswordHashProviderFactory implements PasswordHashProviderFactory {
+
+    public static final String ID = "argon2";
+    public static final String HASH_LENGTH_KEY = "hashLength";
+    public static final String MEMORY_KEY = "memory";
+    public static final String ITERATIONS_KEY = "iterations";
+    public static final String PARALLELISM_KEY = "parallelism";
+
+    private int hashLength;
+    private int memory;
+    private int iterations;
+    private int parallelism;
+
+    @Override
+    public PasswordHashProvider create(KeycloakSession session) {
+        return new Argon2PasswordHashProvider(hashLength, memory, iterations, parallelism);
+    }
+
+    @Override
+    public void init(Config.Scope config) {
+        hashLength = config.getInt(HASH_LENGTH_KEY, 32);
+        memory = config.getInt(MEMORY_KEY, 7168);
+        iterations = config.getInt(ITERATIONS_KEY, 5);
+        parallelism = config.getInt(PARALLELISM_KEY, 1);
+    }
```

Comment:
I see that the provider's hash configurations are not provided via the hash iterations in the password policy, as the `Argon2PhasswordHashProvider#policyCheck` doesn't call `policy.getHashIterations();`. I assume this is intended, but might deserve a comment in the code and in the release notes. 

To document the options available for Argon2, I'd recommend to implement `getConfigMetadata()` so it will auto-generate the documentation in https://www.keycloak.org/server/all-provider-config

Reply:
Yer, with password hashing iterations currently defaulting to 210K this would wreck loads of problems, so for now not making it configurable at all through the realm.

I can add to ConfigMegaData for sure.